packages/store/src/patterns/patternMatchers.js

```diff
@@ -44,6 +43,9 @@ const patternMemo = new WeakSet();
 
 // /////////////////////// Match Helpers Helpers /////////////////////////////
 
+/** For forward references to `M` */
+let MM;
```

Comment:
There's really no better way to do this with some `const [M, matchRemotableHelper,...] = [....`

Reply:
Ran into annoying problems trying to do this. IMO not worth delaying the merge for this. I am proceeding.